CoreScriptsRoot/Modules/Stats/StatsAnnotatedGraph.lua

```diff
@@ -155,13 +155,45 @@ function StatsAnnotatedGraphClass:SetStatsAggregator(aggregator)
   
   self._aggregator = aggregator
   
-  if (self._aggregator ~= nil) then
-    self._listenerId = aggregator:AddListener(function()
-        self:_updateValue()
-    end)
+  self:OnVisibilityChanged()
+end
+
+function StatsAnnotatedGraphClass:_maybeStopListening()
+  if (self._aggregator == nil) then
+    return
+  end
+  
+  if (self._listenerId == nil) then 
+    return
+  end
+  
+  self._aggregator:RemoveListener(self._listenerId)
+  self._listenerId = nil  
+end
+
+function StatsAnnotatedGraphClass:_maybeStartListening()
+  if (self._aggregator == nil) then
+    return
+  end
+  
+  if (self._listenerId ~= nil) then 
+    return
   end
   
-  self:_updateValue()
+  self._listenerId = self._aggregator:AddListener(function()
+      self:_updateValuesAndRender()
+  end)  
+end
+
+
+
+function StatsAnnotatedGraphClass:OnVisibilityChanged()
+  if StatsUtils.PerformanceStatsShouldBeVisible() then
+    self:_maybeStartListening()
+    self:_updateValuesAndRender()
+  else
+    self:_maybeStopListening()
```

Comment:
the maybe seems ambiguous. To the person calling this function we will definitely stop listening when we call this, just the function itself might not have to do anything as it already is not listening. Can we call it just stopListening and startListening since the maybe seems like an implementation detail?